src/journal/journald-server.c

```diff
@@ -326,6 +326,10 @@ static int system_journal_open(Server *s, bool flush_requested) {
                  * Perform an implicit flush to var, leaving the runtime
                  * journal closed, now that the system journal is back.
                  */
+
+                 FIXME: is this still desired if we fail to open the
+                 persistent journal ?
```

Comment:
`server_flush_to_var` calls `system_journal_open`, i.e. there's a loop. It looks like `server_flush_to_var` should be split in two: the inner part which doesn't check any conditions but just does the flush, and it would be called from here, but only if we succeeded to open the journal. The outer part would call `system_journal_open` and check conditions and would be called from other places where we currently call `server_flush_to_var`.